hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -215,6 +215,24 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 
 	// 1. - Reconcile conditions according to current state of the world.
 
+	// Validate the HostedCluster cluster config is valid.
+	// This config is used to set the proxy config for ignition.
+	gConfig, err := globalconfig.ParseGlobalConfig(ctx, hcluster.Spec.Configuration)
+	if err != nil {
+		setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               string(hyperv1.NodePoolValidHostedClusterConditionType),
+			Status:             corev1.ConditionFalse,
+			Message:            err.Error(),
+			Reason:             hyperv1.NodePoolValidationFailedConditionReason,
+			ObservedGeneration: nodePool.Generation,
+		})
+		log.Info("Invalid cluster config for HostedCluster")
+		return reconcile.Result{}, nil
+	}
+	removeStatusCondition(&nodePool.Status.Conditions, string(hyperv1.NodePoolValidHostedClusterConditionType))
```

Comment:
Wouldn't it be better to keep the condition but set it to `true`? Otherwise if the HO doesn't run for whatever reason it makes it look like an invalid config was valid